### ISSUE TITLE
Child windows created on libui applications do not receive some events (Windows)

### ISSUE THREAD

**Issue by rzumer** [andlabs/libui#477]
The event filtering mechanism of libui seems to discard events aimed at windows that are children (or in my experience, children of children) of the libui main HWND. I have a library linked to a libui application which uses the Windows API to spawn extra windows, and then further spawns extra text field "windows" (via `CreateWindowsEx` with the `EDIT` type). These text fields can be navigated with the arrow keys if there is text in them, and they can be edited via copying and pasting using the context menu, but whatever events (`WM_CHAR`?) associated with the keyboard text inputs are not received anymore because, I guess, they are passed to the main HWND instead and then discarded.

The only solution I have right now is to comment out the filtering mechanism, but an upstream solution would be nice. And if I could get an explanation for why this filtering is necessary I would also be interested, since the application on which I did my testing runs just fine without it.

**Comment by andlabs** [comment]
To narrow it down, does it go away if you remove the `areaFilter()` call in `windows/main.cpp`? If not, then the problem is the event plumbing that happens elsewhere in the codebase.

**Comment by rzumer** [comment]
Yes it does, though as expected, merely removing it breaks some of the event handling logic in the application I tested it on. In my case I was able to resort to counting ancestors and only filtering the event if the parent window is up to a 3rd ancestor of the HWND targeted by the event, and obviously that won't work in many cases.
